packages/components/src/utils/flatten-children.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  ReactNode,
+  ReactChild,
+  Children,
+  isValidElement,
+  cloneElement,
+} from 'react';
+import { isFragment } from 'react-is';
+
+/**
+ * Similar to [React's built-in `Children.toArray` method](https://reactjs.org/docs/react-api.html#reactchildrentoarray),
+ * this utility takes children and returns them as an array for introspection or filtering.
+ *
+ * Different from `Children.toArray`, it will flatten arrays and `React.Fragment`s into a regular, one-dimensional
+ * array while ensuring element and fragment keys are preserved, unique, and stable between renders.
+ *
+ * Copied from https://github.com/grrowl/react-keyed-flatten-children (since ESM import doesn't work with the module)
+ */
+export const flattenChildren = (
+  children: ReactNode,
+  depth: number = 0,
+  keys: (string | number)[] = []
+): ReactChild[] =>
+  Children.toArray(children).reduce((acc: ReactChild[], node) => {
```

Comment:
why ReactChild[] is called acc?

Reply:
I copied the code ... `acc` is short for `accumulator` this is what the first argument of `.reduce` is called if you look at the signature. This is the thing that is passed to every callback and returned as a result of the reduce.

Reply:
I was really tempted to not use reduce here because it usually makes the code harder to read ...